torch/_higher_order_ops/utils.py

```diff
@@ -739,17 +740,56 @@ def _tensor_version(t) -> Optional[int]:
         def _tensor_storage(t) -> StorageWeakRef:
             return StorageWeakRef(t._typed_storage())
 
+        def _get_shape_env(
+            fake_args,
+        ) -> Optional[torch.fx.experimental.symbolic_shapes.ShapeEnv]:
+            # detect_fake_mode requires there could be only one active fake mode. This
+            # restricts the usage of this function because the global TracingContext
+            # has a persistent fake mode but fake tensors can be created
+            # outside of the tracing context (e.g. in testing).
+            # Instead, we just look at fake_args fake tensor mode
```

Comment:
umm, why not activate the tracing context?

Reply:
The problem is more about there're fake tensors created under a fake mode that's different from the current activated tracing context's fake mode.

Somehow 1. dynamo has a fake mode in output graph and set it to the tracing context [here](https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/output_graph.py#L405-L412) 2. there is the other backend_fake_mode allocated [here](https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/output_graph.py#L1491-L1497) , which causes mixed fake mode error when we check_input_alias_and_mutation in gen_schema in functionalization key. Both of them seem to refer to the same tracing_context lol